src/sql/workbench/services/executionPlan/common/interfaces.ts

```diff
@@ -33,4 +33,9 @@ export interface IExecutionPlanService {
 	 * Get execution plan file extensions supported by the provider.
 	 */
 	getSupportedExecutionPlanExtensionsForProvider(providerId: string): string[];
+
+	/**
+	 * Get all supported execution plan file extensions across different registered providers.
+	 */
+	getAllSupportedExecutionPlanExtensions(): string[];
```

Comment:
I'd suggest just renaming `getSupportedExecutionPlanExtensionsForProvider` to `getSupportedExecutionPlanExtensions` and have an optional providerId (if undefined then return all)